Tests/SwiftStdlibTests/CollectionExtensionsTests.swift

```diff
@@ -141,4 +141,37 @@ final class CollectionExtensionsTests: XCTestCase {
         XCTAssertEqual([1, 2, 3, 4, 5].average(), 3)
         XCTAssertEqual([Int]().average(), 0)
     }
+    
+    func testAdjacentPairsWithOddNumberOfElementsInCollection() {
+        var pairs: [(Int, Int)] = []
+        for (first, second) in [1, 2, 3].adjacentPairs() {
+            pairs.append((first, second))
+        }
```

Comment:
Nit: Would that be simpler?
```suggestion
        let pairs = Array([1, 2, 3].adjacentPairs())  
```

Reply:
Valid for other occurrences as well if applicable.

Reply:
Done. 
replaced with the above kind of code wherever applicable.
Thanks for the suggestion!